docs/gen-ai/gen-ai-metrics.md

```diff
@@ -0,0 +1,157 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: Generative AI metrics
+--->
+
+# Semantic Conventions for Generative AI Metrics
+
+**Status**: [Experimental][DocumentStatus]
+
+The conventions described in this section are specific to Generative AI
+applications.
+
+**Disclaimer:** These are initial Generative AI client metric instruments
+and attributes but more may be added in the future.
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [Generative AI Operations](#generative-ai-operations)
+  - [Metric: `gen_ai.token.usage`](#metric-gen_aitokenusage)
+  - [Metric: `gen_ai.operation.duration`](#metric-gen_aioperationduration)
+
+<!-- tocstop -->
+
+## Generative AI Operations
+
+The following metric instruments describe Generative AI operations. An
+operation may be a request to an LLM, a function call, or some other
+distinct action within a larger Generative AI workflow.
+
+### Metric: `gen_ai.token.usage`
+
+This metric is [required][MetricRequired] when an operation involves the usage
```

Comment:
I wonder what we should do about streaming:

OpenAI does not count prompt tokens at all, i.e. there will be no input tokens. 
OpenAI returns chunk per token and instrumentation can count them easily.

I think we can either:
1. postpone adding streaming support into this metric. Tell it's required for non-streaming cases and is not supported for streaming ones. Create an  issue to track the work there
2.  Add something like this to the metric description.

Metric is required for non-streaming calls and recommended for streaming if instrumentation can efficiently obtain information about input and/or output tokens count.

For example, if GenAI system returns usage information in the streaming response, it SHOULD be used. Or if GenAI system returns each token independently, instrumentation SHOULD count number of output tokens and record the result.

If instrumentation cannot efficiently obtain number of input and/or output tokens, it MAY allow users to enable offline token counting. Otherwise it MUST NOT report usage metric.

Reply:
also given that usage is not always available, we should make this metric recommended